### DISCUSSION TITLE
Question about Halt

### DISCUSSION THREAD

**Discussion by molove** [phil-barrett/grblHAL-teensy-4.x#29] [Q&A] [answered]
Hi
I'm starting to wire up my cnc controller and have a quick question about halt. Does engaging halt kill power to the relays as well as halting all moves? 
Thanks
Piers

**Comment by phil-barrett** [comment, marked as answer]
It definitely drops power to the Spindle, Vac, Mist and Flood relays.  I believe it drops power to Aux1-3 but you should verify that with a recent version of grblHAL.

You probably know this but Halt is purely a grblHAL construct.  The Teensy does not actually have a halt pin that resets it so the Halt pin on the BoB is just another input.   The main implication of is that you might want to have a separate emergency power cut off switch.  Generally halt works well as an EStop, though.

A side benefit of the Teensy's lack of reset is that grblHAL does not lose position information.  Useful when you use halt as an EStop.  I've restarted several jobs with no issues.

**Comment by molove** [comment]
Thanks Phil, it was an EStop I wanted to use it. There will be a power off for the whole system too.
